### PR TITLE
Fix voice silently failing to connect to channel

### PR DESCRIPTION
**Pull request by Lunarmagpie**
### Summary
`hikari.impl.VoiceImpl.connect_to` would silently fail (awaits function forever) to connect to a voice channel if the guild_id or channel_id was wrong. This PR adds timeout when connecting.`hikari.errors.VoiceError` is raised when this timeout is reached. A `timeout` kwarg is added if users want to change the timeout. The default timeout is 5s.

I could not find how they handle errors in [this section](https://discord.com/developers/docs/topics/gateway#update-voice-state) of the API docs but it looks like they just dont respond with anything so I decided a timeout would be the best implementation.

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

**Review by davfsa — COMMENTED** [review]
Thank you for the bug investigation and the fix!

File: hikari/api/voice.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ async def connect_to(
         *,
         deaf: bool = False,
         mute: bool = False,
+        timeout: typing.Optional[int] = 3,
```
[the remark this answers is not in the document]
Comment (Lunarmagpie) [reply hikari/api/voice.py:89]: I changed the default to 5. Wasn't sure if 3 or 5 would be better anyway lol.

File: changes/1242.bugfix.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix hikari.impl.VoiceImpl.connect_to silently failing if the guild or channel id is wrong.
```
[the remark this answers is not in the document]
Comment (Lunarmagpie) [reply changes/1242.bugfix.md]: Actually I dont think this is accurate. This bug also occurs when you provide a channel that does exist but it is not a voice channel.
I'm going to change it to "Fix `hikari.impl.VoiceImpl.connect_to` silently failing if the guild or channel do not exist or the channel is not a voice channel by providing a timeout."

File: hikari/api/voice.py [hunk 3]
```diff
@@ -106,6 +107,15 @@ async def connect_to(
...
             Defaulting to `builtins.False`, if `builtins.True`, the client will
             enter the voice channel muted (thus unable to send audio).
+        timeout: typing.Optional[builtins.int]
+            Defaulting to `5`, The amount of time to wait before erroring when
+            connecting to the voice channel. If timeout is `None` there will be
+            no timeout.
+
+        !!! warning
+            If timeout is `None`, this function will be awaited forever if an
+            invalid `guild_id` or `channel_id` is provided.
```
Comment (davfsa) [inline hikari/api/voice.py:117]: Would need to be indented too :)

File: changes/1242.bugfix.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davfsa) [reply changes/1242.bugfix.md]: > Actually I dont think this is accurate. This bug also occurs when you provide a channel that does exist but it is not a voice channel.

Could be changed to " if the guild or voice channel do not exist by providing a timeout.`, should cover both cases

[the remark this answers is not in the document]
Comment (Lunarmagpie) [reply changes/1242.bugfix.md]: I think that description is a bit ambiguous on what happens if the channel exists but its not a voice channel. Bikeshedding though so ill change if you really want me to.

**Review by davfsa — APPROVED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**